.github/workflows/pull-request-ci.yml

```diff
@@ -0,0 +1,30 @@
+name: PullRequest check
+
+on:
+  pull_request:
+    branches:
+      - main
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+
+    steps:
+      - uses: actions/checkout@v4
+      - name: Use Node.js 20
+        uses: actions/setup-node@v4
+        with:
+          node-version: 20
+      - uses: oven-sh/setup-bun@v1
```

Comment:
Installed packages should be cached for speeding up the worlflow imo
e.g.
```yaml
jobs:
  build:
    runs-on: ubuntu-latest
    steps:
      - name: checkout
        uses: actions/checkout@v3

      - name: setup-node
        uses: actions/setup-node@v3
        with:
          node-version: 18
          cache: npm
          cache-dependency-path: ./next/package-lock.json

      - name: cache-node-modules
        uses: actions/cache@v3
        id: node_modules_cache_id
        env:
          cache-name: cache-node-modules
        with:
          working-directory: ./next
          path: '**/node_modules'
          key: ${{ runner.os }}-build-${{ env.cache-name }}-${{ hashFiles('**/package-lock.json') }}

      - name: npm-install
        if: ${{ steps.node_modules_cache_id.outputs.cache-hit != 'true' }}
        run: npm i
        working-directory: ./next
```

Reply:
Same as above